### PR TITLE
merchantdash

### PR DESCRIPTION
**Pull request by PhiMed**
- Feat/tests: MerchantItemsIndex
- Feat/test: MerchantItems show story
- Formatting
- Feat/tests: MerchantItem Update
- Feat: Flash alert

### CODE REVIEW

File: app/views/merchant_items/edit.html.erb [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+<%= form_with url: "/merchants/#{@item.merchant_id}/items/#{@item.id}", method: :patch, local: true do |f| %>
+  <%= f.label :name %>
+  <%= f.text_field :name, value: @item.name, required: true  %><br>
+
+
+  <%= f.label :description %>
+  <%= f.text_field :description, value: @item.description, size: 75, required: true %><br>
+
+  <%= f.label :unit_price %>
+  <%= f.number_field :unit_price, value: @item.unit_price.to_s, required: true %>
```
Comment (yosoynatebrown) [inline app/views/merchant_items/edit.html.erb:11]: Nice job setting default values in a form

File: spec/features/merchant_items/show_spec.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+    visit "/merchants/#{@merchant.id}/items/#{@item_1.id}"
+
+    within('#update_link') do
+      click_link
+    end
+
+    fill_in 'Name', with: "Moazinite Rabbit"
+    fill_in 'Description', with: "25mmx25mm Moazinite Rabbit"
+    fill_in 'Unit price', with: "4500"
```
Comment (AlexMMcConnell) [inline spec/features/merchant_items/show_spec.rb:28]: Add sad path test where certain fields are *not* filled in and the form can still successfully submit.

Comment (AlexMMcConnell) [reply spec/features/merchant_items/show_spec.rb:28]: Also add sad path where non numeric input is entered

Comment (GregoryJFischer) [reply spec/features/merchant_items/show_spec.rb:28]: sad path test added for not filled in fields, non-numeric input is disallowed by html and validations which is tested

**Pull request merged**